### PR TITLE
Application Database Download

### PR DESCRIPTION
**Pull request by qiarie**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

**Engineer Checklist**
- [ ] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [x] I have added any strings visible on UI components to the `strings.xml` file
- [ ] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [x] I have built and run the FHIRCore app to verify my change fixes the issue and/or does not break the app 
- [ ] I have checked that this PR does NOT introduce **breaking changes** that require an update to **_Content_** and/or **_Configs_**? _If it does add a sample here or a link to exactly what changes need to be made to the content._

**Code Reviewer Checklist**
- [ ] I have verified **Unit tests** have been written for any new feature(s) and edge cases
- [ ] I have verified any strings visible on UI components are in the `strings.xml` file
- [ ] I have verifed the [CHANGELOG.md](./CHANGELOG.md) file has any notable changes to the codebase
- [ ] I have verified the solution has been implemented in a configurable and generic way for reuseable components
- [ ] I have built and run the FHIRCore app to verify the change fixes the issue and/or does not break the app

### CODE REVIEW

**Comment by dubdabasoduba** [comment]
@qiarie Please add the link the SDK issue here

**Comment by qiarie** [comment]
@dubdabasoduba [SDK issue #2360 - Ability to Dump Application's Database](https://github.com/google/android-fhir/issues/2360)

**Comment by dubdabasoduba** [comment]
@qiarie can we have this also done for the `main` branch

File: android/quest/src/main/java/org/smartregister/fhircore/quest/ui/usersetting/UserSettingViewModel.kt [hunk 1]
```diff
@@ -179,4 +191,63 @@ constructor(
   fun dismissInsightsView() {
     viewModelScope.launch { unsyncedResourcesMutableSharedFlow.emit(listOf()) }
   }
+
+  private fun copyDatabase(context: Context, onCopyCompleteListener: () -> Unit) {
+    viewModelScope.launch(dispatcherProvider.io()) {
+      try {
+        val passphrase = DBUtils.getEncryptionPassphrase("fhirEngineDbPassphrase")
```
Comment (ndegwamartin) [inline android/quest/src/main/java/org/smartregister/fhircore/quest/ui/usersetting/UserSettingViewModel.kt:198]: Is this the password that encrypts the extracted DB? 

Perhaps we should prompt the user to enter a specific password that can be shared with the troubleshooter for more security. 

If its a heavy lift I'm ok with this being in _v2_ or the version that goes to _main_

Comment (qiarie) [reply android/quest/src/main/java/org/smartregister/fhircore/quest/ui/usersetting/UserSettingViewModel.kt:198]: This passphrase is what FHIR engine uses to encrypt the app's Sqlite DB. The class & methods that handle this in the SDK are private and we had to do a workaround to be able to decrypt the app DB from FHIRCore.
There's a conversation around having the DB decrypted in the SDK as documented [in SDK issue #2360](https://github.com/google/android-fhir/issues/2360). That implementation is what is meant to be used in main.

The password we are using to encrypt zipped DB file after download is `{username}_{first 8 characters of the practitioner_id}`.

For purposes of the release to unblock eCHIS, we had gone with a derivative password to simplify the download and sharing. We can include the prompt for a password in the next iteration using the SDK.

**Review by ageryck — APPROVED** [review]

**Pull request merged**